Fixed Combobox small sizing

it-vegard opened the pull request:
### Description

Remove extra vertical padding and fixed height for combobox input, to make the small variant the same size as TextField
Ref. Slack discussion: https://nav-it.slack.com/archives/C7NE7A8UF/p1709640219782299

New heights are 32px for small and 48px for medium/default

### Change summary

- Removed vertical padding and reduced size of input on small variant of combobox.

it-vegard committed:
Remove extra vertical padding and fixed height for combobox input, to make the small variant the same size as TextField and so on.

it-vegard committed:
Merge branch 'main' into combobox-small-sizing

it-vegard committed:
no message

it-vegard committed:
More robust combobox styling - using tokens and proper padding instead of min-height for sizing

it-vegard committed:
More robust combobox styling using tokens

it-vegard committed:
Remove gap between options in FilteredOptions

it-vegard committed:
Merge branch 'main' into combobox-small-sizing

HalvorHaugan commented on @navikt/core/css/form/combobox.css:
```diff
@@ -1,18 +1,35 @@
   flex-direction: row;
   align-items: center;
   justify-content: space-between;
+  min-height: initial;
```
Gjør `min-height: initial` noe?

HalvorHaugan commented on @navikt/core/css/form/combobox.css:
```diff
@@ -198,17 +211,11 @@
-  height: 1.5rem;
+  width: var(--__ac-combobox-icon-size);
+  height: var(--__ac-combobox-icon-size);
   pointer-events: none;
```
Litt off topic, men lurer på hvorfor denne har `pointer-events: none`?

it-vegard replied:
Her er nok et tilfelle der squashing av commits ødelegger litt for å forstå koden. Eventuelt et case der vi burde hatt inline-kommentarer. Fant ingen nytte for den nå, så tok bort. Kan ha vært koblet til tidligere ustabilitet med "dobbeltklikk" på åpne/lukke-knappen, men at det er fikset på en mer robust måte nå? 🤷

HalvorHaugan commented on @navikt/core/css/form/combobox.css:
```diff
@@ -226,16 +233,15 @@
-  width: 1.5rem;
+  height: var(--__ac-combobox-icon-size);
+  width: var(--__ac-combobox-icon-size);
 }
```
Du kan jo eventuelt vurdere å slå sammen alle selektorene som setter svg-størrelse til `.navds-combobox__wrapper svg`

HalvorHaugan replied:
Oops, dette overstyrer også X-ikonet på chipsa 🙈

it-vegard replied:
Det gjorde det. 🙈 Har gått tilbake til separate selectorer, men samlet, slik at størrelsen bare settes ett sted.

HalvorHaugan commented on @navikt/core/css/form/combobox.css:
```diff
@@ -330,7 +324,8 @@
   border-left: 4px solid var(--ac-combobox-list-item-selected-hover-border-left, var(--a-border-focus));
-  padding-left: calc(var(--a-spacing-3) - 4px);
+
+  /* padding-left: calc(var(--a-spacing-3) - 4px); */
```
```suggestion
```

it-vegard committed:
Update @navikt/core/css/form/combobox.css

Co-authored-by: Halvor Haugan <83693529+HalvorHaugan@users.noreply.github.com>

it-vegard committed:
Combine selectors for svg sizes in combobox

Removed pointer-events: none, because I couldn't find a use-case for it. The reason vanished when we squashed commits.

it-vegard committed:
Fix visual regression, where the first option was partially overlapped by the "max selected" message

it-vegard committed:
merge: Merge branch with main

it-vegard committed:
Merge branch 'main' into combobox-small-sizing

it-vegard committed:
Revert yarn lock to match main

it-vegard committed:
A general selector for combobox SVGs had an effect on chips as well, so we need a selector for each SVG we want to set size for.

it-vegard committed:
Properly align items in FilteredOptions. Fixes issue with out-of-alignment checkmarks.

it-vegard committed:
Merge branch 'main' into combobox-small-sizing

HalvorHaugan commented on @navikt/core/css/form/combobox.css:
```diff
@@ -1,18 +1,42 @@
+
+.navds-combobox__button-clear svg,
+.navds-combobox__button-toggle-list svg,
+.navds-combobox__button-list svg {
```
Finnes ikke noe element med klassen `navds-combobox__button-list`. Mente du `navds-combobox__list`?

it-vegard committed:
Merge branch 'main' into combobox-small-sizing

it-vegard committed:
Fix incorrect class selector

it-vegard committed:
This doesn't seem to do anything
